### DISCUSSION TITLE
Auto Update

### DISCUSSION THREAD

**Discussion by jkfoong** [JPersson77/LGTVCompanion#136] [Ideas]
Any plans to include auto-updater instead of downloading the installer when there is update available?

**Comment by JPersson77** [comment]
Hi @jkfoong, I have not given it much thought but I will consider it. Thanks for the suggestion!

**Reply by jkfoong** [reply]
No problem. Not a C++ programmer but C# thus not sure if I could help.
Attached a library as reference.
[https://github.com/VioletGiraffe/github-releases-autoupdater](url)

 Might work on a PR later if I have time to look into C++.

**Reply by JPersson77** [reply]
That library looks like it could be integrated very easily. If you want to have a go at a PR have a look at the code for the daemon. At start of the daemon a thread is spawned running VersionCheckThread(). The thread will post a message back to the daemon message loop when an update is found, which in turn will show the notification, and write over IPC to the server which will also post it in the server log. There is a similar function in the project for the main/options UI. Why I have not put them both in the common module I don't know but I should :) 

I guess a new checkbox would be good "Prompt for automatic update" so there is some minor updates to the preferences read/write function in the common module to be done as well as the options dialog resource, and the initalisation of the options window.
